walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -22,59 +22,187 @@
 import java.util.List;
 
 public final class FindMeetingQuery {
-  
-  private Collection<TimeRange> possibleTimes ;
 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    //throw new UnsupportedOperationException("TODO: Implement this method.");
-    // If no attndees
-    if ((request.getAttendees() == null ) && (events == null) ){
+
+    // Convert collection of events into an arraylist.
+    ArrayList<Event> eventList = new ArrayList<>(events);
+    
+    // Initialise result variable. 
+    Collection<TimeRange> possibleTimes = new ArrayList<>();
+    // Result considering only mandatory attendees.
+    Collection<TimeRange> timesMandatory = new ArrayList<>();
+
+    // Initialise variables for no events and no attendees.
+    Collection<Event> NO_EVENTS = Collections.emptySet();
+    Collection<String> NO_ATTENDEES = Collections.emptySet();
+
+    
+    // Check if there are optional attendee, who is busy and can't attend.
+    for (int i=0; i< eventList.size(); i++) {
```

Comment:
nit: for (Event event : eventList) { is a nicer way of making loops in java, that way you don't have to use eventList.get(i) every time, you can just use event.
Typically we try to avoid using loops with indexes unless we specifically need the index, because it's easier to read